modules/javafx.base/src/main/java/javafx/collections/ModifiableObservableListBase.java

```diff
@@ -308,43 +336,62 @@ public boolean containsAll(Collection<?> c) {
 
         @Override
         public boolean addAll(Collection<? extends E> c) {
+            if (c.isEmpty()) {
+                return false;
+            }
+
             beginChange();
             try {
-                boolean res = sublist.addAll(c);
-                return res;
+                return sublist.addAll(c);
             } finally {
                 endChange();
             }
         }
 
         @Override
         public boolean addAll(int index, Collection<? extends E> c) {
+            if (c.isEmpty()) {
+                if (index < 0 || index > sublist.size()) {
+                    throw new IndexOutOfBoundsException("Index: " + index);
+                }
+
+                return false;
+            }
+
```

Comment:
Shouldn't this always check the index, not only when the collection passed in is empty?

Otherwise code like: `singleElementSubList.addAll(15, List.of("a"))` would be allowed?

```suggestion
            if (index < 0 || index > sublist.size()) {
                throw new IndexOutOfBoundsException("Index: " + index);
            }
            if (c.isEmpty()) {  // implicit null check
                return false;
            }
```